services/httppost/service.go

```diff
@@ -207,6 +210,7 @@ func (s *Service) Handler(c HandlerConfig, l *log.Logger) alert.Handler {
 		endpoint: e,
 		logger:   l,
 		headers:  c.Headers,
+		hc:       &http.Client{Timeout: c.Timeout},
```

Comment:
I like the idea of using `context.WithTimeout` on each request, but if you choose a different approach and use a custom `http.Client` be sure to add a transport that uses `ProxyFromEnvironment`.

```
&http.Client{
    Timeout: c.Timeout,
    Trasport: &http.Transport{
			Proxy: http.ProxyFromEnvironment,
		},
}
```